### PR TITLE
[AF-544] Allow close a expanded dock via API

### PR DESCRIPTION
**Pull request by ederign**

### CODE REVIEW

**Comment by ederign** [comment]
@paulovmr thanks for the review! fixed!

**Comment by porcelli** [comment]
Wondering about the purpose of the "DeSelect" feature... 

What about toggle feature? If this is about close/expand.. why not use setExpand or just close() or hide?

**Comment by ederign** [comment]
@porcelli deselect and expand API will be used by bookmarkable URLs. Even if we use to expand or close without the reference for the dock, we should pass with dock area we are wanna close.
 
Is that ok for you?

File: uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/UberfireDocksImpl.java [hunk 1]
```diff
@@ -114,6 +112,13 @@ public void expand(UberfireDock dock) {
     }
 
     @Override
+    public void deselect(UberfireDock dock) {
+        if (docksBars.isReady()) {
+            docksBars.deselect(dock);
```
Comment (porcelli) [inline uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/UberfireDocksImpl.java]: what if... is not ready? aren't we loosing the call?

Comment (porcelli) [inline uberfire-extensions/uberfire-simple-docks/uberfire-simple-docks-client/src/main/java/org/uberfire/client/docks/UberfireDocksImpl.java]: what about add to a registry of after initialization/ready?

**Pull request closed without merging**